index.js

```diff
@@ -23,6 +23,10 @@ const pThrottle = ({limit, interval}) => {
 
 	return function_ => {
 		const throttled = function (...args) {
+			if (!throttled.isEnabled) {
+				return Promise.resolve(function_.apply(this, args));
```

Comment:
```suggestion
				return (async () => function_.apply(this, args))();
```

because otherwise, if `function_` throws it will end up as a synchronous exception instead of a promise rejection.

Can you also add a test for this case?